nixos/doc/manual/configuration/linux-kernel.chapter.md

```diff
@@ -133,3 +133,8 @@ This section was moved to the [Nixpkgs manual](https://nixos.org/nixpkgs/manual#
 It's a common issue that the latest stable version of ZFS doesn't support the latest
 available Linux kernel. It is recommended to use the latest available LTS that's compatible
 with ZFS. Usually this is the default kernel provided by nixpkgs (i.e. `pkgs.linuxPackages`).
+
+:::{.note}
+Using non-LTS releases with ZFS is *not supported* because it is likely to break.
```

Comment:
I think this wording is a bit strong. It makes it seem like ZFS will malfunction with a non-LTS kernel. But the only breakage is an error at eval time, so the user can pick another kernel or otherwise address the issue.

I think it's reasonable to say NixOS "supports" any kernel+ZFS combination that doesn't fail to eval, to the extent that "support" means the kernel module should work. If I understand correctly, the "support" that's being dropped is just for those cases that would cause eval failures anyway.

Reply:
"support" generally implies that it can be expected to keep working, and in this case that's not true.  Somebody could have hardware that's too new for the LTS kernel, but the only newer kernels supported by ZFS might have gone EOL and been removed from Nixpkgs.

Reply:
The problem is that ZFS works with the latest non-LTS kernel for like half the year and doesn't for the other half.

At install time, it might just happen to be the time of the year where the newest non-LTS kernel is supported by ZFS, so the user installs with no errors and all is good but then a few months later it's EOL'd and ZFS won't support the next kernel for at least a few months. The user is then left with a system that doesn't eval anymore which I'd consider breakage.

The problem is not how we handle the breakage induced by kernel EOLs but rather that users are not initially made aware that this breakage can happen and that it's actually very likely to happen. The purpose is mostly to set the expectations right from the beginning; inform users to do the right thing before their systems break and not have a bad awakening a few months down the line.

As @maralorn mentioned for instance, he wouldn't have gotten himself into the predicament where he will soon have to choose between filesystem or WiFi working had this warning existed when he installed NixOS on his new laptop last week and still had the choice of not using ZFS.

Reply:
I understand all that, and I agree with the general premise of this PR. I am commenting only on this one line:

> Using non-LTS releases with ZFS is *not supported* because it is likely to break.

As a user, this line makes me think that using ZFS with a non-LTS kernel could cause crashes or data corruption on NixOS. That's not the case, since any actual incompatibility should be caught at eval. I think this line could be rephrased so that it doesn't imply more than intended.

I've used ZFS with a vendor kernel for a couple years, so I understand the pitfalls. I think we can simultaneously say that using non-LTS kernels with ZFS is not a supported config while also saying that running non-LTS kernels with ZFS can be expected to work _as long as the build successfully evals_.

Reply:
> this line makes me think that using ZFS with a non-LTS kernel could cause crashes or data corruption on NixOS

Right, now I get it. Yes, that's indeed not the intention but I also didn't want to re-iterate the entire eval warning here.

We should find a wording that captures the kind of breakage that is to be expected here.

> My question is: what does this statement add that isn’t already said in the previous paragraph plus the new eval warning (plus the NixOS wiki)?

Nothing. I only added it to the manual for completeness' sake.

While the Wiki is sort of official now, I feel like the manual is the more appropriate place to say "this is how it is" whereas the Wiki is more of a "this is how you do x"/"this is how you work with x".

> Have there been a lot of issues getting opened? If so, I haven’t seen too many. More so that folks are confused, rather than demanding that it works.

Preventing said confusion pre-emptively is the one and only purpose.